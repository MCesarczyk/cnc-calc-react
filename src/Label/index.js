import React from "react";
import "./style.css";

const Label = ({ control, binding, text, input }) => (
    <label className="label">
        <span className="labelInnerText">
            {control}<label for={binding}>{text}:</label>
        </span>
        {input}
    </label>
);

export default Label;