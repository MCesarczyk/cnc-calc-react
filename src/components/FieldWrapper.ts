import styled from 'styled-components';

export const FieldWrapper = styled.div`
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
`;
