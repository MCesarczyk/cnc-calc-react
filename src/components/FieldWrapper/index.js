import styled from "styled-components";

export const FieldWrapper = styled.div`
    display: flex;
    gap: 0.5rem;
    align-items: center;
    justify-content: space-between;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
        gap: 0.25rem;
    }
`;