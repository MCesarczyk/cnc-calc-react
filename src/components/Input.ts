import styled from 'styled-components';

export const Input = styled.input`
  border: solid 1px ${({ theme }) => theme.color.primary};
  border-radius: 5px;
  padding: 5px;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    width: 100%;
  }

  &:invalid {
    background-color: ${({ theme }) => theme.color.invalid};
  }

  &:disabled {
    background-color: ${({ theme }) => theme.color.disabled};
  }
`;
