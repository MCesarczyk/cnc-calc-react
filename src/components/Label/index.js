import styled from "styled-components";

export const Label = styled.label`
    text-align: right;

    @media (max-width: 480px) {
        align-self: flex-start;
    }
`;