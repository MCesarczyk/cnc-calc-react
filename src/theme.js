export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}