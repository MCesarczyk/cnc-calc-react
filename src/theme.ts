export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
