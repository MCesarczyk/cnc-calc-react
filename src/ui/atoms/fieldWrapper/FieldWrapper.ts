import styled from 'styled-components';

export type FieldWrapperVariants = 'DEFAULT';

export type Palette<TVariants extends string, TStyles> = {
  [variant in TVariants]: TStyles;
};

interface FieldWrapperProps {
  $variant?: FieldWrapperVariants;
}

export const FieldWrapper = styled.div<FieldWrapperProps>`
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
`;

FieldWrapper.defaultProps = {
  $variant: 'DEFAULT',
  children: 'Sample content',
};

FieldWrapper.displayName = 'FieldWrapper';
