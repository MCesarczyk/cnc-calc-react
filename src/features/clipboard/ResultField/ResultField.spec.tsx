import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { FieldWrapper } from 'components/FieldWrapper';
import { Label } from 'components/Label';
import { theme } from 'theme';
import { Input } from 'ui/atoms/input/Input';

test('ResultField component rendering properly', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <FieldWrapper>
        <Label />
        <Input />
      </FieldWrapper>
    </ThemeProvider>,
  );

  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
