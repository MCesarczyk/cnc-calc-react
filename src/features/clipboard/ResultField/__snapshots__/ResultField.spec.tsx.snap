// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultField component rendering properly 1`] = `
.c0 {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;
}

.c1 {
  text-align: right;
}

.c2 {
  border: solid 1px #1e3246;
  border-radius: 5px;
  padding: 5px;
}

.c2:invalid {
  background-color: beige;
}

.c2:disabled {
  background-color: #ddd;
}

@media (max-width: 480px) {
  .c0 {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
}

@media (max-width: 480px) {
  .c1 {
    align-self: flex-start;
  }
}

@media (max-width: 480px) {
  .c2 {
    width: 100%;
  }
}

<div
  className="c0"
>
  <label
    className="c1"
  />
  <input
    className="c2"
  />
</div>
`;
